gobblin-metrics/src/main/java/gobblin/metrics/metric/filter/NotificationTimerFilter.java

```diff
@@ -0,0 +1,24 @@
+package gobblin.metrics.metric.filter;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+
+import gobblin.metrics.MetricContext;
+
+
+/**
+ * Implementation of {@link MetricFilter} that ignores metrics with name
+ * {@link MetricContext#GOBBLIN_METRICS_NOTIFICATIONS_TIMER_NAME}.
+ *
+ * <p>
+ *   The {@link MetricContext#GOBBLIN_METRICS_NOTIFICATIONS_TIMER_NAME} metric is used to profile how long it takes to
+ *   send {@link gobblin.metrics.notification.Notification}s; this is useful for profiling the gobblin-metrics library.
+ *   So most of the time this {@link com.codahale.metrics.Timer} can be filtered out.
+ * </p>
+ */
+public class NotificationTimerFilter implements MetricFilter {
```

Comment:
Can we make this more generic? We might add more profiling metrics in the future, and the same filter should work for all of them. Maybe have all profiling metrics share a prefix, and this filter should check for that prefix?